doc/engsys_resource_management.md

```diff
@@ -0,0 +1,94 @@
+## Resource management guidelines
+
+This document contains guidelines for creating and managing Azure resources in the Azure SDK Engineering System
+subscriptions. EngSys has automation that will delete resources based on the criteria outlined below. This is to
+improve our overall security and keep us in compliance with company policies.
+
+  * [Managing Dev Resources in the Playground Subscription](#managing-dev-resources-in-the-playground-subscription)
+     * [Resource Groups](#resource-groups)
+     * [Role Assignments](#role-assignments)
+  * [Managing CI resources in the test subscription](#managing-ci-resources-in-the-test-subscription)
+     * [Resource Groups](#resource-groups-1)
+     * [Role Assignments](#role-assignments-1)
+
+
+## Managing Dev Resources in the Playground Subscription
+
+This section applies to resource groups located in the `Azure SDK Developer Playground` subscription.
+
+Currently the automation will inspect resource groups and role assignments only.
+
+### Resource Groups
+
+The automation will delete all resource groups in the playground subscription DAILY, unless they meet one of the following criteria:
+
+- The resource group starts with a valid Microsoft alias within `microsoft.com` or `ntdev.microsoft.com`, followed by an
+  optional hyphen and extra characters.
+    - Valid group name examples: `myalias`, `myalias-test-foobar` for `myalias@microsoft.com`
+- The resource group contains a tag with the name `owners` where the value is a csv formatted string that contains at
+  least one valid Microsoft alias within `microsoft.com` or `ntdev.microsoft.com`. This convention should only be used
+  when it is not possible to name the resource group with an alias (for example, inner groups auto-created by a resource
+  provider like AKS).
+    - Valid owner tag examples: `owners: myalias`, `owners: myalias,anotheralias,lastalias`
+- The resource group contains a tag with the name `DeleteAfter` and an ISO8601 formatted date value, where the date is
+  not greater than 7 days in the future.
+    - If the `DeleteAfter` value is in the past, or greater than 7 days in the future, it will be deleted.
```

Comment:
We should probably add validation `[ValidateInRange(1, 7*24)]` to the `-DeleteAfterHours` parameter of the TestResources scripts to help codify this.

Reply:
Good idea, though this should only apply to the playground sub, unless we think we also want to disable longer delete after periods in the testing sub? I can't see why anyone would need a longer one.

Reply:
> I can't see why anyone would need a longer one.

That's what I was figuring as well. If they need it longer, maybe should be in the static-test-resources RG.

Reply:
I sometimes run adhoc tests that go for a couple of weeks. Should I not?

Reply:
You can, but you should extend the lifetime.

Reply:
@richardpark-msft if you have tests in the stress cluster, they will be excluded as the parent resources are marked as static. Otherwise you should extend the lease as @heaths says, and this doc will detail multiple mechanisms by which to do that.

Reply:
> We should probably add validation `[ValidateInRange(1, 7*24)]` to the `-DeleteAfterHours` parameter of the TestResources scripts to help codify this.

I'll do this separately so I'm not bound to the eng/common sync in this PR.